### ISSUE TITLE
Launch planning

### ISSUE THREAD

**Issue by kytrinyx** [exercism/bash#4]
_Note: the launch checklist has been made obsolete by a brand new Launch Guide: https://github.com/exercism/docs/blob/master/language-tracks/launch/README.md_

Let's keep the general discussions around what needs to be done here in this issue, and open new, actionable issues for next steps.

**Comment by aleveille** [comment]
Just checking on the state of this issue:
- Bash is missing at least 3 problems. I'm new to Exercism, but quite proficient in Bash. Is there any problems that are idiomatic of Exercism that we want exercises for? Or should I go for exercises that I think will make the more sense for bash?
- I see that some documentation have been written, but perhaps we can improve it a little bit 😃 
- I believe the logo has been taken care of, right?

**Comment by kenden** [comment]
@aleveille I just picked problems in other tracks that followed the ones we have already in xbash, and ported them. There might be a better way, some easier problems in, say, Python, can be much harder in bash but then we can just "mark them harder" in config.json.

The documentation can always be improved :)
Especially how to set up the local devenv to contribute https://github.com/exercism/xbash/issues/1
But that could apply to the [installation of bash](https://github.com/exercism/xbash/blob/master/docs/INSTALLATION.md) too. Especially on macs, the version installed by default is older (it's the BSD version, the GNU version is newer). That means the example for [gigasecond](https://github.com/exercism/xbash/tree/master/exercises/gigasecond) does not work on macs.

It would be nicer to clarify too if the exercises can use linux utilities (date, grep, awk, sed...) or if it should be just "pure bash".

**Comment by kytrinyx** [comment]
> Bash is missing at least 3 problems.  

We have a page that lists all the exercises that are missing in bash: http://exercism.io/languages/bash/contribute

> Or should I go for exercises that I think will make the more sense for bash?

I think this is the best approach.

> The documentation can always be improved :)

Haha, yes!

**Comment by kytrinyx** [comment]
@aleveille @kenden we have enough exercises to launch now, but we still need some help with both the user-facing documentation in `docs/` and the contributing documentation (README.md).

I think we still also need to make a decision about whether or not linux utilities are available to us (my first thought is "yes", since a huge part of bash scripting is getting familiar with the various utilities).

**Comment by Smarticles101** [comment]
Does the documentation still need some work before we can launch @kytrinyx?

**Comment by kenden** [comment]
Hm, if we say linux utilities are available to us, what versions? macOS uses old versions of most tools (sed, base64, grep, date, cat...) which behave slightly differently. 

Most of the time it's possible to make a program work on linux and macOs by finding the option that works on both platforms (ex `base64 --decode` instead of `base64 -D`.

Or we could require to install the GNU version, but it's not always easy. For `sed` for example, updating is not so trivial: https://stackoverflow.com/questions/30003570/how-to-use-gnu-sed-on-mac-os-x.

Or we could run in a docker image / virtualmachine where we know the version installed, but that makes getting started harder.

I vote for: only bash, no extra utilities installed, no posix/sh/zsh compatibility required.
(Plus, sed/awk are complicated enough that we could make a track just for them!)

**Comment by kytrinyx** [comment]
@kenden You make good points.

I don't know enough bash to have a strong opinion here, so I'm going to let you all hash out the pros/cons and make a suggestion for what you think is wise.

**Comment by kytrinyx** [comment]
I've updated this issue to point to the new launch guide. In short: let's keep discussions about what to do next to this issue, and use these discussions + the guide as a jumping off point to opening new, actionable issues.

**Comment by budmc29** [comment]
> I vote for: only bash, no extra utilities needed, no posix/sh/zsh compatibility required.

That sounds good, the default tools can get the job done and if we keep the tests POSIX compliant we should be fine (extra practice for that will always help the contributors).

> (Plus, sed/awk are complicated enough that we could make a track just for them, so allowing them is a bit like cheating!)

I will definitely be up to help maintain a sed and/or awk track, since I always regret not spending more time with that, especially when I have to do tedious tasks that take minutes when I know that those tools will get the job done in seconds.

**Comment by budmc29** [comment]
What is needed to have Bash as language available for V2?
Is the weak documentation the only blocker?

**Comment by kytrinyx** [comment]
Documentation is the big blocker.

> I vote for: only bash, no extra utilities needed, no posix/sh/zsh compatibility required.

If we decide on this we need to go through the existing exercises and make sure they follow these guidelines. We'll also need to document this in the README for contributors/new maintainers.

It would probably be worth going through the launch guide step by step for the bash track to make sure we have all the bases covered:

https://github.com/exercism/docs/blob/master/language-tracks/launch/README.md

If you do, then please open issues on the docs repo whenever the guide is confusing or is missing anything. I wrote it all last weekend, and by the end of it my brain was pretty fried.

**Comment by budmc29** [comment]
Will read it and start working on whatever I can in preparation.

**Comment by budmc29** [comment]
#### Getting Bash in Exercism V2

Decisions:
  - [ ] Programming language variants (discussion opened)
  - [x] Language name and track ID
  - [x] Installation and dependencies
  - [x] Testing framework

Landing Page - Prepare the track:
  - [ ] Language track icon (requested)
  - [ ] Introductory copy
  - [ ] User-facing documentation
  - [x] Maintainer bios (both maintainers have show_on_website: false, is that an issue?)

@kytrinyx let me know if I missed anything from this checklist

**Comment by kytrinyx** [comment]
I think this looks right. It's no problem that `show_on_website` is false.

**Comment by sjwarner-bp** [comment]
Hi all, I've contributed a few times here and would love to help get this language track up and running for exercism v2!

Where can I start? I'm more than happy to trawl through documentation, I just want a couple of pointers of what and where :slightly_smiling_face: 

I think the config.json has been sorted (there are lots of open issues to add topics, but I think all exercises have at least one topic assigned).

**Comment by kytrinyx** [comment]
@sjwarner-bp I've added you to the list of maintainers for the track. I don't actually know the state of things here, unfortunately. That might reasonably be the most useful thing you could help figure out 🌷

**Comment by sjwarner-bp** [comment]
Thanks for that @kytrinyx - I'll try and work out the state of things here and do any necessary tidy-up. I don't feel like it's too far away from being ready 😄

**Comment by Smarticles101** [comment]
@sjwarner-bp and @kytrinyx 
I think that the `config.json` definitely needs a little fixing up as in the order of exercises and difficulties for most should be re-done. There are very little exercises, and only one of them has a difficulty higher than 2, which doesn't seem quite right. Maintainer-wise, I've been the only one I've seen around for a while. My bash skills are only mediocre, so most pull requests look fine to me as long as there's nothing clearly wrong with the code. It would also be worth running exercise tests in travis. Also, some of the exercises use `awk` which doesn't comply with the plain bash decision that was made in #88

**Comment by sjwarner-bp** [comment]
@Smarticles101 thanks for the update! I was also going to suggest the `config.json` needs a little more work, but I'm happy to pick that up in the near future. I'd be interested to hear your thoughts on it 🙂 

Ok, that's good to know - I don't often like to merge my own submissions, as a second pair of eyes is always welcome, though I can see this might be a little more difficult here! I'll try and help out here wherever I can.

**Comment by kytrinyx** [comment]
> There are very little exercises, and only one of them has a difficulty higher than 2, which doesn't seem quite right.

Yeah, it feels like if we could get an extra handful (5? 10?) exercises, that would make the track feel more substantial at launch.

**Comment by sjwarner-bp** [comment]
I'm working on implementing `grains` at the moment! Hopefully we can get a few more exercises in and all of the documentation done :slightly_smiling_face:

**Comment by budmc29** [comment]
I tried to get involved with Bash a while ago but looked like when there were decisions to be made things were not moving forward.

It would be nice to have someone set the direction and start working towards that, for example if you see https://github.com/exercism/bash/issues/88 is still opened and nothing was decided.

@kytrinyx what are your thoughts on this?

**Comment by kytrinyx** [comment]
@budmc29 I need people to take charge and make decisions—preferably 3 people together.

I am not the right person to set the direction, since I am not conversant in Bash.

With respect to decision making, I find that it helps to separate the _debate_ phase from the _decision_ phase. This tends to make it easier to dig into and identify the questions that are actually being discussed and the trade-offs that need to be made.

**Comment by sjwarner-bp** [comment]
@budmc29 Looking over the state of #88, I think that a decision has been made (everyone seems to be in agreement) but it hasn't been properly actioned yet. I think that the only issue moving forwards is actioning it. How would you suggest us 'completing' this particular objective - it seems to be a bit of a blocker. Is this to simply document the decision, or is there more code that needs reworking (I'm relatively new here) :slightly_smiling_face:

**Comment by budmc29** [comment]
@sjwarner-bp you are right, but I don't know either how are we supposed to move it forward (it applies to pretty much every decision or open PR).

As @kytrinyx suggested we need 3 maintainers for the track.
There are currently 3 entries is the maintainers.json files, yourself, @adelcambre and @kotp.

I know for sure you are active and seems that you have time to participate in the track (correct me if I'm wrong), we also need to see if the same is for @adelcambre and @kotp.

@Smarticles101 how is your availability, are you able to be a maintainer as well?

Given that all the current maintainers are available and maybe if @Smarticles101 want to join to, then I think we can easily progress through issues or decisions when we have a say from all the maintainers (and we know how can make those decisions).

I'd be happy to get involved with this track as well, bash is such a nice language to know as a developer.

**Comment by Smarticles101** [comment]
@budmc29 I'm actually in the maintainer team for bash, I just haven't added myself to the maintainers.json file lol.

Currently I have a bit of spare availability, but as of recent I'm not available a lot. I try to look at pull requests when I can although it can take a while for me to review them after they're made.

There are actually 5 maintainers on the team... @adelcambre, @kotp, @kenden, myself, and @sjwarner-bp. I've never seen @adelcambre in the time I've contributed, and I haven't seen @kotp in a while and I haven't seen @kenden since September I believe.

As for #88 @budmc29 and @sjwarner-bp, it affects both code and documentation, but we should continue that conversation there :)

**Comment by kytrinyx** [comment]
@budmc29 I've added you to the team as well.

I chatted with @adelcambre recently, and he's had other things on his mind and in his life ✨ 🌻 ✨

**Comment by budmc29** [comment]
@Smarticles101 sorry about that :D.

@kytrinyx thank you, glad to be part of the team.

I'm don't have much free time these days with the holidays but I'll jump in and help with what I can when I get back to work.

@sjwarner-bp thank you for all the hard work, you're doing a great job on moving things forward and taking action, you motivated me to get back into this track. Keep up the good work :clap:

**Comment by budmc29** [comment]
#### Getting Bash in Exercism V2

Decisions:
  - [x] Programming language variants
  - [x] Language name and track ID
  - [x] Installation and dependencies
  - [x] Testing framework

Landing Page - Prepare the track:
  - [ ] Language track icon (requested but still waiting for response)
  - [ ] Introductory copy
  - [ ] User-facing documentation
  - [x] Maintainer bios

**Comment by sjwarner-bp** [comment]
@budmc29 - do we not have a language track icon already? Please correct me if I am mistaken, but this point references the image seen [here](http://exercism.io/languages/bash/launch)?

Introductory copy definitely needs some improvement, I'll set about doing that in my free time.

User-facing documentation has the bare bones there pretty much I think - could do with a little bit more detail in places. Again, something I'd be happy to pick up!

**Comment by sjwarner-bp** [comment]
Also, a little while ago @budmc29 was making changes to tests and example exercises to improve consistency - are there many more exercises that are inconsistent?

**Comment by budmc29** [comment]
> @budmc29 - do we not have a language track icon already? Please correct me if I am mistaken, but this point references the image seen here?

That's the official icon, but we need a fancy icon to match the style of the new website, see: https://s3-eu-west-1.amazonaws.com/exercism-static/tracks/coffeescript-bordered-turquoise.png

> Also, a little while ago @budmc29 was making changes to tests and example exercises to improve consistency - are there many more exercises that are inconsistent?

I have no idea, but I was planing to add the .meta folder to all of them, I'll have a look then.

**Comment by sjwarner-bp** [comment]
Ah, that makes more sense - thanks :+1:

**Comment by kotp** [comment]
exercism/website-icons#3

**Comment by budmc29** [comment]
That's good.

The only things left to make the track ready are:

  - [ ] Introductory copy
  - [ ] User-facing documentation

**Comment by budmc29** [comment]
That's good.

The only things left to make the track ready are:

  - [ ] Introductory copy
  - [ ] User-facing documentation

And of course it would be nice to get more exercises, but that can be sorted later.

**Comment by sjwarner-bp** [comment]
> And of course it would be nice to get more exercises, but that can be sorted later.
I think I've got `scrabble-score` in progress...

**Comment by budmc29** [comment]
Resolved in #168
